server/apps/publish/archive_publish.py

```diff
@@ -595,6 +583,34 @@ def _update_headline_sequence(self, doc):
         if doc.get(SEQUENCE):
             doc['headline'] = '{}={}'.format(doc['headline'], doc.get(SEQUENCE))
 
+    def _validate_package_contents(self, package, takes_package):
+        """
+        If the item passed is a package this function will ensure that the unpublished content validates and none of
+         the content is locked by other than the publishing session, also do not allow any killed or spiked content
+        :param package:
+        :param takes_package:
+        :raises: Validation exceptions if the validation fails
+        """
+        # Ensure it is the sort of thing we need to validate
+        if package[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE and not takes_package and self.publish_type == 'publish':
```

Comment:
I think we have a constant defined for publish types (created recently in the other pull request)?